packages/drivers/driver-web-cache/src/FluidCache.ts

```diff
@@ -117,10 +120,35 @@ export class FluidCache implements IPersistedCache {
 		});
 	}
 
+	private async openDb() {
+		if (this.db === undefined) {
+			this.db = await getFluidCacheIndexedDbInstance(this.logger);
+			// Need to close the db on version change if opened.
+			this.db.addEventListener("versionchange", (ev) => {
+				this.db?.close();
+				this.db = undefined;
+			});
+			this.db.addEventListener("close", (ev) => {
+				clearTimeout(this.dbCloseTimer);
+				this.dbCloseTimer = undefined;
+				this.db = undefined;
+			});
+			// Schedule db close after 30s.
+			if (this.dbCloseTimer === undefined) {
+				this.dbCloseTimer = setTimeout(() => {
+					this.db?.close();
+					this.db = undefined;
+					this.dbCloseTimer = undefined;
+				}, 30000);
```

Comment:
>30000 [](http://example.com/codeflow?start=7&length=5)

why 30 secs ? Is it random ?

Reply:
Yes, random for now. Just want to keep it open for sometime. Should be enough for load/catching up etc.

Reply:
This may cause issues for the host app, in particular we've seen this mean that signout hangs.

Reply:
There are two possible solutions here I feel like:
- decrease timeout substantially. Even 1-2 seconds would result in over 99% of cases reusing opened DB. I'd start with 2 seconds, but also record if db was reused or not on each cache call.
- Not use timeouts but bake a notion of "session" into cache layer, such that driver can control precisely duration of session (when to close DB). It's a bit hard to implement it that way, as driver does not fully control whole load sequence - Loader does, but I think we need to make certain changes in driver interface (irrespective of that change) that will make it possible in the future.

One thing to add - it might be great to have explicit "release all resources" call on cache. Driver can use it when driver is disposed (for example error is hit). But driver could also call this API in future where we know we are don with load sequence.

Reply:
I was thinking that we should have some kind of DB session pattern that is required for all DB interactions to ensure that the session reacts to DB deletions (`versionchange`) events appropriately and backed with enforcement or unit tests to make sure the driver is not blocking DB deletions. I'm not sure if the notion of "session" you're describing overlaps or is higher in the hierarchy, but I'm thinking either approach would help us to ensure DB sessions behave consistently.

Reply:
I'm not sure I follow comments about all sessions or impact on deletion. Could you please clarify that?
Before & after change have same behavior - DB is left opened for a while (i.e. code is async between open and close).
What is changed here is that duration of DB being opened is longer. Is this the primary concern? I.e. are we saying that keeping DB opened longer is bad because it will cause more trouble, and count wise we can go over some thresholds?
Or something else?
(Let's assume we will use 1-2 seconds, not 30 seconds for this discussion)

Reply:
RE session: I meant container load session. I.e. we know that we need to make couple calls to cache during load.
The problem is that driver does not control it today, it's controlled by Loader layer and it would be weird hard to add notion of session here.

Reply:
> Before & after change have same behavior - DB is left opened for a while (i.e. code is async between open and close).

I don't think this is true for the before case, this is precisely what [the previous PR](https://github.com/microsoft/FluidFramework/pull/14315/files) I created changed, ensuring DBs are closed after they are open and used in a synchronous fashion. This is because DBs were left open and were blocking DB deletion in other functions and browser tabs. This PR seems to fundamentally change that behavior, reverting in part to leaving DBs open for some amount of time after they are used.

Reply:
There is await in between open & close. Even if it takes 100ms, other activity could be scheduled that hit exactly same condition - DB is opened and in use. I do not see a difference here, other than amount of time we are talking about,

Reply:
Sorry, yes, you are correct. The bigger problem before my PR is that there were code paths that would never close the DB, so yes, the amount of time is the biggest concern (and ensuring we don't add any new code paths that would never close the DB as was the case previously).